packages/next/src/generators/application/lib/create-application-files.ts

```diff
@@ -28,11 +28,27 @@ export function createApplicationFiles(host: Tree, options: NormalizedSchema) {
 
   generateFiles(
     host,
-    join(__dirname, '../files'),
+    join(__dirname, '../files/common'),
     options.appProjectRoot,
     templateVariables
   );
 
+  if (options.appDir) {
```

Comment:
You should also update `next.config.js` to include `experimental: { appDir: true }`